src/main/java/com/tabnine/binary/BinaryRequestFacade.java

```diff
@@ -38,16 +39,21 @@ public BinaryRequestFacade(BinaryProcessRequesterProvider binaryProcessRequester
             return result;
         } catch (TimeoutException e) {
             binaryProcessRequesterProvider.onTimeout();
+            Sentry.captureException(e);
         } catch (ExecutionException e) {
             if (e.getCause() instanceof TabNineDeadException) {
                 binaryProcessRequesterProvider.onDead(e.getCause());
             }
 
-            Logger.getInstance(getClass()).warn("Tabnine's threw an unknown error during request.", e);
+            String message = "Tabnine's threw an unknown error during request.";
+            Logger.getInstance(getClass()).warn(message, e);
+            Sentry.captureException(e, message);
```

Comment:
I do not like these Sentry calls all over the code.
I think it is better to use the logger, whenever an error/warning is reported, send to sentry too.

They have integration with the populr logging frameworks - here's [an example](https://docs.sentry.io/platforms/java/guides/log4j2/)

If it is a problem to make it work with intellij, we can always create a wrapper around the logger and report to sentry from there.